src/en/manhuascan/src/eu/kanade/tachiyomi/extension/en/manhuascan/ManhuaScan.kt

```diff
@@ -0,0 +1,334 @@
+package eu.kanade.tachiyomi.extension.en.manhuascan
+
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.util.Calendar
+
+class ManhuaScan : ConfigurableSource, ParsedHttpSource() {
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val name = "ManhuaScan"
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    override val baseUrl = getMirror()
+
+    override val client by lazy {
+        network.cloudflareClient.newBuilder()
+            .rateLimit(2)
+            .build()
+    }
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    // ============================== Popular ===============================
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/popular${page.getPage()}", headers)
+
+    override fun popularMangaSelector(): String = ".manga-list > .book-item"
+
+    override fun popularMangaFromElement(element: Element): SManga = SManga.create().apply {
+        thumbnail_url = element.selectFirst(".thumb img")?.imgAttr()
+        element.selectFirst(".title a")!!.run {
+            title = text()
+            setUrlWithoutDomain(attr("abs:href") + "#${text()}")
```

Comment:
saving title in `url` field, why not access it directly from `SManga.title` when needed like in `chapterListRequest`?

Reply:
Good point